### ISSUE TITLE
MacOS builds with incorrect name for tarball

### ISSUE THREAD

**Issue by Rasbats** [Rasbats/shipdriver_pi#475]
As per Zulip ...

MacOS has tarball name ShipDriver-3.2-darwin-wx315-10.13.6-tar…
Should it be darwin-wx32?

[https://cloudsmith.io/~opencpn/repos/shipdriver-alpha/packages/](https://cloudsmith.io/~opencpn/repos/shipdriver-alpha/packages/)

**Comment by leamas** [comment]
yes. Will look into it

**Comment by Rasbats** [comment]
Correct naming but does not import into the MacOS alpha 5.7.1+f6a1838b2. The latest o-charts plugin does not import either. Both report incompatible.

**Comment by Rasbats** [comment]
5.7.1 has had a lot of recent changes. Wonder if a new build for the Mac is needed?

**Comment by leamas** [comment]
> 5.7.1 has had a lot of recent changes. Wonder if a new build for the Mac is needed?

Yes. If it still reports incompatible, please submit the logfile using `--loglevel debug`

**Comment by Rasbats** [comment]
Made a MacOS OpenCPN build from the latest OpenCPN master. Still the same issue. Do we have the same situation as Windows. The .sh for the OpenCPN build is using a download of wxWidgets from [opencpn.org](opencpn.org). Again the latest o-charts plugin does not import or ShipDriver. Both report incompatible. ShipDriver is using a homebrew version of wxWidgets. About to get the log.

**Comment by Rasbats** [comment]
Log from my MacOS build:
[debug.pdf](https://github.com/Rasbats/shipdriver_pi/files/10211959/debug.pdf)

**Comment by leamas** [comment]
https://github.com/OpenCPN/OpenCPN/issues/2875

**Comment by Rasbats** [comment]
Compiled MacOS OpenCPN from https://github.com/OpenCPN/OpenCPN/commit/f0993d586793da66f665732e547e385132f3a665

Tried importing ShipDriver _darwin-wx32_ tarball but still getting "plugin incompatible". 

Compiled MacOS ShipDriver using same wxWidgets as OpenCPN. Still "plugin incompatible".

**Comment by Rasbats** [comment]
Ok ... log coming!

**Comment by Rasbats** [comment]
[debug2.pdf](https://github.com/Rasbats/shipdriver_pi/files/10217544/debug2.pdf)

**Comment by leamas** [comment]
hrmpf... If you look at the metadata file in the tarball, what is the `<target>` value?

**Comment by leamas** [comment]
Thinking aloud: We have this code in plugin_handler.cpp:
```
 static const std::vector<std::string> simple_abis = {"msvc", "msvc-wx32",
  "darwin", "darwin-wx32", "android-armhf", "android-arm64"};

  auto compatOS = CompatOs::getInstance();
  Host host(compatOS);
  Plugin plugin(metadata);

  auto found = std::find(simple_abis.begin(), simple_abis.end(), plugin.abi());
  if (found != simple_abis.end()) {
    bool ok = plugin.abi() == host.abi();
    wxLogDebug("Returning %s for %s", (ok ? "ok" : "fail"), host.abi());
    wxLogDebug(" ");
    return ok;
  }

```

If target indeed is _darwin-wx32_ it should match the corresponding entry in simple_abis nad we should get a debug log like _Returning ok for darwin-wx32_ or so. But it seems that the plugin metadata isn't in that list. Strange...

**Comment by leamas** [comment]
Here is something very, very rotten Using `--loglevel debug` there should be a lot of lines like _Plugin: setting up, name: ...._  etc. but I cannot find them. Is this the complete log?

**Comment by leamas** [comment]
First, I think whatr we see is the entries when parsing the catalog. In my case, the look like this:
```
4:27:37.237   DEBUG plugin_handler.cpp:185 Host: init: abi: fedora-x86_64, abi_version: 36, major ver: 36
14:27:37.237   DEBUG plugin_handler.cpp:162 Plugin: setting up, name: AIS Radar view
14:27:37.237   DEBUG plugin_handler.cpp:163 Plugin: init: abi: debian-arm64, abi_version: 11, major ver: 11
14:27:37.237   DEBUG plugin_handler.cpp:370 Plugin compatibility check Final: REJECTED: AIS Radar view

```
They should definitely  look similar on macos. 

ALso: when I look for shipdriver in the log I only find an entry which I **think** is related to the shipdriver entry in the catalog.  That is, I don't see any trace from importing  a tarball.

**Comment by Rasbats** [comment]
Back exploring again. I will try another debug log.

**Comment by Rasbats** [comment]
In the tarball _target_ is _darwin-wx32_.
_target-version_ is 10. In OpenCPN 5.6.2 plugins the _target-version_ is 10.13.6
Can _target-version_ cause a problem?

**Comment by Rasbats** [comment]
And another debug log:
[debug3.pdf](https://github.com/Rasbats/shipdriver_pi/files/10219166/debug3.pdf)

**Comment by leamas** [comment]
> Can target-version cause a problem?

No. As you can see in the code snippet above, if we match on darwin-wx32 the version is not checked.

The problem is the missing debug output which should be there like in my log. Partly because we need that info, and partly because we cannot really trust a build which behaves like this.

The good part is this:

```
14:55:51.581 DEBUG plugin_handler.cpp:311 Plugin compatibility check Final:
REJECTED: ShipDriver
```

which seems to be from the failed import. But then again, we need the other output like in my log to figure out what's happening.

One thing: If you start the dreaded   Options | Plugins |Settings.  In the Compatibility panel, what is the Active setting?

**Comment by leamas** [comment]
Another OpenCPN bug: https://github.com/OpenCPN/OpenCPN/pull/2879

Can you apply this to your build and see what happens?

**Comment by Rasbats** [comment]
>what is the Active setting?

_Disabled_

Using _CatalogExpert=1_ has no effect on menu options.

**Comment by leamas** [comment]
Something is just plain wrong here, either with the build or with the environment.  You can work around this by setting the environment variable OPENCPN_COMPAT_TARGET to _darwin-wx32_ before starting openpcn and try again. This should override the default host target which we know is wrong, see PR above

**Comment by Rasbats** [comment]
Brilliant. ShipDriver from Cloudsmith imported successfully.
But ... Getting the plugin incompatible notice in the same way as we experienced with Windows.
But ... I had compiled my own MacOS version of ShipDriver using the opencpn.org download of wxWidgets instead of homebrew.
This version imports and runs successfully.
Time for another discussion of which wxWidgets to use. Believe Pavel is using the homebrew version.

**Comment by leamas** [comment]
This is the same wxWidgets, they are both 3.2. And in runtime you don't use your own anyway, you use the wxWidgets bundled with OpenCPN. So, this is not an issue. 

See https://github.com/OpenCPN/OpenCPN/discussions/2797#discussioncomment-4373532.

The SD bulid chain is set up to use homebrew.

**Comment by Rasbats** [comment]
Tried to modify OpenCPN build to use wxWidgets from the homebrew cellar but it could not find the INCLUDE_LIBRARIES. Never mind ... the log from my standard build of OpenCPN and standard ShipDriver is here:

[debug4.pdf](https://github.com/Rasbats/shipdriver_pi/files/10220654/debug4.pdf)

**Comment by Rasbats** [comment]
Finally got an OpenCPN build to work with homebrew wxWidgets. Standard ShipDriver from Cloudsmith works fine.

Debug log is looking more like you expect. The installer used homebrew wxWidgets as well.

[debug5.pdf](https://github.com/Rasbats/shipdriver_pi/files/10221895/debug5.pdf)

**Comment by leamas** [comment]
Time to close this one after merging #479?

**Comment by Rasbats** [comment]
Now closing.
